### ISSUE TITLE
white page when activating plugin

### ISSUE THREAD

**Issue by jimlongo56** [wpsharks/comet-cache#721]
Hi, I've been trying to use your plugin on a cPanel Centos6 VPS with PHP5.5

But in most cases when I activate comet cache, I start getting dropped connections in the admin screens,  or white pages on the front end.

I just enabled debugging for a moment on one site, and when trying to view a front facing page (which displays a white page) get the following errors written to the debug.log.  

```
[21-Mar-2016 12:53:34 UTC] PHP Warning:  include_once(/home/therapy/public_html/wp-content/plugins/comet-cache/src/includes/stub.php): failed to open stream: No such file or directory in /home/therapy/public_html/wp-content/advanced-cache.php on line 23
[21-Mar-2016 12:53:34 UTC] PHP Warning:  include_once(): Failed opening '/home/therapy/public_html/wp-content/plugins/comet-cache/src/includes/stub.php' for inclusion (include_path='.:/opt/cpanel/ea-php55/root/usr/share/pear') in /home/therapy/public_html/wp-content/advanced-cache.php on line 23
```

Thanks for any help.

_More information: I've had just rebuilt this server before trying your plugin. Whereas before I was using mod_security with the suPHP handler, I'm now using mod_ruid2 with the DSO php handler_

**Comment by raamdev** [comment]
@jimlongo56 Please follow these [manual uninstallation instructions](https://cometcache.com/kb-article/how-do-i-uninstall-comet-cache/#toc-86754ab8) and then reinstall Comet Cache. That should solve the problem. (The issue is that your `wp-content/advanced-cache.php` file is outdated and needs to be force-replaced by Comet Cache with an updated copy.)

**Comment by jimlongo56** [comment]
Hi, thanks for the response.

I had tried those manual uninstallation instructions a few times and always the same result.
Noticed that every time I installed the cache folder kept being recreated as zencache.
Thought that perhaps because I couldn't delete the plugin from the admin panel, it wasn't cleaning up after itself fully.

What I had to do to completely remove all traces was also to go to the database and delete all wp_options 
`WHERE`option_value`LIKE  '%comet%'`
`WHERE`option_value`LIKE  '%zen%'`
`WHERE`option_name`LIKE  '%comet%'`
`WHERE`option_name`LIKE  '%zen%'`

Also deleted all `WHERE`option_value`LIKE  '%transient%'`

Once I did that then I could install cleanly and function as expected.

Thanks again.

**Comment by raamdev** [comment]
@jimlongo56 Thanks for the update. For future reference, if you want to have ZenCache / Comet Cache completely remove even the database options, you can deactivate plugin deletion safeguards, save options, and then Deactivate + Delete the plugin from **Dashboard → Plugins**. When you Delete the plugin with Plugin Deletion Safeguards disabled, ZenCache / Comet Cache will remove even the database options.

![2016-03-21_13-24-58](https://cloud.githubusercontent.com/assets/53005/13927512/5a067e1e-ef68-11e5-9c42-85f0271e7354.png)

**Comment by jimlongo56** [comment]
Good to know. 

My problem was that I could only delete the plugin by FTP so it was not deleting the options.

**Comment by jimlongo56** [comment]
@raamdev I installed this it's working fine, and then next morning, BOOM! it all stops, white page, can't login, nothing.
I move the comet-cache plugin folder by ftp up one level to wp-content, and I have access again.
I've looked there are no errors, it just stops serving the cached pages.

Once I move the plugin folder it will serve content as normal, I put the folder back, and white pages.
**the response is 503 (Service Unavailable)**

Any idea what might be causing that? varnish? my server build of Apache/PHP - specific module that might be imcompatible.

When it works it's fantastic, and i'm certainly going to purchase the Pro version if I can get it to reliably work.

**Comment by raamdev** [comment]
@jimlongo56 A white screen indicates an error. If you just see a white screen and no error that means one of two things:
- PHP error logging has been disabled. You should enable it; see https://codex.wordpress.org/Debugging_in_WordPress
- Apache is throwing an error. In this case, you need to look at your Apache error log. 

We need to see what error is being thrown.

**Comment by jimlongo56** [comment]
From access logs It appears that the site stopped serving pages at the same time the cache was rebuilt
Last log entries about 5:22

 Comet Cache file built for (http://spadinatherapycentre.com/) in 0.80812 seconds, on: Mar 22nd, 2016 @ 5:24 pm UTC.

**Comment by jimlongo56** [comment]
[22-Mar-2016 23:01:01 UTC] PHP Warning:  Invalid argument supplied for foreach() in /home/therapy/public_html/wp-content/plugins/comet-cache/src/includes/closures/Ac/ObUtils.php on line 199

**Comment by raamdev** [comment]
PHP Warnings would not generate a white screen. 

Also, have you reviewed [Testing Comet Cache in a Clean WordPress Installation](https://cometcache.com/kb-article/testing-comet-cache-in-a-clean-wordpress-installation/)?

**Comment by jimlongo56** [comment]
I have another site (same VPS, different account) that is working fine.

**Comment by jimlongo56** [comment]
There's nothing in the Apache error log.

**Comment by raamdev** [comment]
> I have another site (same VPS, different account) that is working fine.

So that would clearly indicate this isn't a Comet Cache issue but an issue with the configuration of that account that is causing problems with WordPress caching in general. I suggest comparing what's different between those two accounts and going from there.

**Comment by jimlongo56** [comment]
i commented these lines from the .htaccess, can't imagine why they'd matter in this instance, yet doing that the plugin works again.

```
AddType audio/mp4 .m4a
AddType video/mp4 .m4v
AddType video/ogg .ogv 
AddType video/mp4 .mp4 
AddType video/webm .webm
```

Then I immmediately uncommented them as to prove they are causing the issue, yet it still works.

So I'm stumped.  I'm have a feeling this won't be the last time.

**Comment by jimlongo56** [comment]
I've also turned off Varnish on the problematic site, that's the only other difference I can see so far.

I hope.

**Comment by jimlongo56** [comment]
FYI, the issue seems to be directly related to my Apache build.

As stated in the beginning of the ticket I had rebuilt Apache recently using cPanel's EasyApache4.
I had just upgraded from EA3, so it was a new build profile.
The profile was using **mod_ruid2 + DSO handler for PHP** which was new for me.

I changed the build profile back to what it had been previously, using **mod_security and su_ExecPHP**.
This completely eliminated all my difficulties. I installed the lite version on a few test sites, and it worked in all cases as expected.

Hopefully something to consider if others have this same problem.

Anyway have bought a pro license and configured it to use my Cloudfront CDN, works fantastically.
Thanks for all your help.
